deepchem/transformers/__init__.py

```diff
@@ -456,3 +457,56 @@ def transform_array(self, X, y, w):
 
   def untranform(self, z):
     print("Cannot undo CoulombBinarizationTransformer.")
+
+class CDFTransformer(Transformer):
+  """Histograms the data and assigns values based on sorted list."""
+  """Acts like a Cumulative Distribution Function (CDF)."""
+  def __init__(self, transform_X=False,
+               transform_y=False,
+               bins=2,
+               dataset=None):
+    self.dataset = dataset
+    self.transform_X = transform_X
+    self.transform_y = transform_y
+    self.bins = bins
+
+  def transform(self, dataset):
+    """Performs CDF transform on data."""
+    X, y, w, ids = dataset.to_numpy()
+    w_t = w
+    ids_t = ids
+    if self.transform_X:
+      X_t = self.get_cdf_values(X)
+      y_t = y
+    if self.transform_y:
+      y_t = self.get_cdf_values(y)
+      X_t = X
+    # TODO (rbharath): Find a more elegant solution to saving the data?
+    # shutil.rmtree(dataset.data_dir)
+    # os.makedirs(dataset.data_dir)
+    # Dataset.from_numpy(dataset.data_dir, X_t, y_t, w_t, ids_t)
+    
+    return X_t, y_t, w_t, ids_t
+
+  def get_cdf_values(self, array): 
+    cols = len(array[0,:])
+    rows = len(array[:,0])
+    array_t = np.zeros((rows,cols))
+    parts = rows/self.bins
+    hist_values = np.zeros((rows,1))
+    sorted_hist_values = np.zeros((rows,1))
+    for row in xrange(rows):
+      if np.remainder(self.bins,2)==1:
+        hist_values[row,0] = np.floor(np.divide(row,parts))*1/(self.bins-1)
+      else:
+        hist_values[row,0] = np.floor(np.divide(row,parts))*1/self.bins
+    order = np.argsort(array, axis=0)
```

Comment:
I think this is wrong. Note that each column in array has a different sort order! (since each feature is from a different distribution). So, the order only makes sense for a column at a time. Try moving the order computation into the for-loop below.